pkg/stores/partition/listprocessor/processor.go

```diff
@@ -0,0 +1,287 @@
+	pageParam     = "page"
+	revisionParam = "revision"
+)
+
+// ListOptions represents the query parameters that may be included in a list request.
+type ListOptions struct {
+	ChunkSize  int
+	Resume     string
+	Filters    []Filter
+	Sort       Sort
+	Pagination Pagination
+	Revision   string
+}
+
+// Filter represents a field to filter by.
+// A subfield in an object is represented in a request query using . notation, e.g. 'metadata.name'.
+// The subfield is internally represented as a slice, e.g. [metadata, name].
+type Filter struct {
+	field []string
+	match string
+}
+
+// String returns the filter as a query string.
+func (f Filter) String() string {
+	field := strings.Join(f.field, ".")
+	return field + "=" + f.match
+}
+
+// SortOrder represents whether the list should be ascending or descending.
+type SortOrder int
+
+const (
+	// ASC stands for ascending order.
+	ASC SortOrder = iota
+	// DESC stands for descending (reverse) order.
+	DESC
+)
+
+// Sort represents the criteria to sort on.
+// The subfield to sort by is represented in a request query using . notation, e.g. 'metadata.name'.
+// The subfield is internally represented as a slice, e.g. [metadata, name].
+// The order is represented by prefixing the sort key by '-', e.g. sort=-metadata.name.
+type Sort struct {
+	primaryField   []string
+	secondaryField []string
+	primaryOrder   SortOrder
+	secondaryOrder SortOrder
+}
+
+// String returns the sort parameters as a query string.
+func (s Sort) String() string {
+	field := ""
+	if s.primaryOrder == DESC {
+		field = "-" + field
+	}
+	field += strings.Join(s.primaryField, ".")
+	if len(s.secondaryField) > 0 {
+		field += ","
+		if s.secondaryOrder == DESC {
+			field += "-"
+		}
+		field += strings.Join(s.secondaryField, ".")
+	}
+	return field
+}
+
+// Pagination represents how to return paginated results.
+type Pagination struct {
+	pageSize int
+	page     int
+}
+
+// PageSize returns the integer page size.
+func (p Pagination) PageSize() int {
+	return p.pageSize
+}
+
+// ParseQuery parses the query params of a request and returns a ListOptions.
+func ParseQuery(apiOp *types.APIRequest) *ListOptions {
+	chunkSize := getLimit(apiOp)
+	q := apiOp.Request.URL.Query()
+	cont := q.Get(continueParam)
+	filterParams := q[filterParam]
+	filterOpts := []Filter{}
+	for _, filters := range filterParams {
+		filter := strings.Split(filters, "=")
+		if len(filter) != 2 {
+			continue
+		}
+		filterOpts = append(filterOpts, Filter{field: strings.Split(filter[0], "."), match: filter[1]})
+	}
+	// sort the filter fields so they can be used as a cache key in the store
+	sort.Slice(filterOpts, func(i, j int) bool {
+		fieldI := strings.Join(filterOpts[i].field, ".")
+		fieldJ := strings.Join(filterOpts[j].field, ".")
+		return fieldI < fieldJ
+	})
+	sortOpts := Sort{}
+	sortKeys := q.Get(sortParam)
+	if sortKeys != "" {
+		sortParts := strings.SplitN(sortKeys, ",", 2)
+		primaryField := sortParts[0]
+		if primaryField[0] == '-' {
```

Comment:
Not quite. The length of the parameter is checked on 120. Then the slice created on 121 must have at least one non-empty string in it. That string is shortened by 1 on 125. At that point, it's possible it could be empty, hence the check on 127.

Reply:
You will still need the check for both primaryField and secondaryField because of the `strings.SplitN(...)`
https://go.dev/play/p/ELlnx4eMRwS